js/main.js

```diff
@@ -0,0 +1,247 @@
+
+    for (var i = 0; i < count; i++) {
+      var locationX = getRandomNumber(mapPinsArea.offsetWidth) + 1;
+
+      // Получаем рандомную координату по Y. Для этого берем минимальное значение (чтобы в любом случае было 130)
+      // И добавляем интервал в 500 для вычисления координаты по У. Т.е. maxY может быть только 630.
+      var locationY = Math.floor(MIN_COORDINATE_Y + (Math.random() * (MAX_COORDINATE_Y - MIN_COORDINATE_Y) + 1));
+      var dataForPins = {
+        'author': {
+          'avatar': 'img/avatars/user0' + (i + 1) + '.png'
+        },
+        'offer': {
+          'title': getRandomElementFromArray(OFFER_TITLES),
+          'address': locationX + ', ' + locationY,
+          'price': getRandomNumber(PRICE),
+          'type': getRandomElementFromArray(HOUSE_TYPES),
+          'rooms': getRandomNumber(MAX_ROOMS),
+          'guests': getRandomNumber(MAX_GUESTS),
+          'checkin': getRandomElementFromArray(CHECK_INS),
+          'checkout': getRandomElementFromArray(CHECK_OUTS),
+          'features': getRandomNumberOfElementsFromArray(FEATURES),
+          'description': getRandomElementFromArray(DESCRIPTIONS),
+          'photos': getRandomElementFromArray(PHOTOS)
+        },
+        'location': {
+          'x': locationX,
+          'y': locationY
+        }
+      };
+
+      dataList.push(dataForPins);
+    }
+    return dataList;
+  }
+
+  // Добавляем функцию по настройке пинов. Клонируем имеющийся темплейт, выбираем в новом темплейте аватар по селектору.
+  // Присваиваем значения left/top/src/alt, возвращаем новый пин
+  function setPinOptions(data) {
+    var newPin = pinTemplate.cloneNode(true);
+    var pinAvatar = newPin.querySelector('img');
+
+    newPin.style =
+      'left: ' + data.location.x + 'px; top: ' + data.location.y + 'px;';
+    pinAvatar.src = data.author.avatar;
+    pinAvatar.alt = data.author.title;
+
+    return newPin;
+  }
+
+  // Добавляем функцию по созданию пинов. Создаем пустой documentFragment, запускаем цикл по генерации пинов в соответствии
+  // с данными из функции setPinOptions, которая обращается к createMocksForData, добавляем их в пустой documentFragment. Вставляем получившийся documentFragment
+  // в map__pins.
+  function generatePins(pinsList) {
+    var fragment = document.createDocumentFragment();
+
+    for (var i = 0; i < pinsList.length; i++) {
+      fragment.appendChild(setPinOptions(pinsList[i]));
+    }
+
+    mapPinsArea.appendChild(fragment);
+  }
+
+  // Запускаем цепочку функций по генерации пинов.
+  generatePins(createMocksForData(OFFERS_NUMBER));
+
+  // Функция по генерации информации для карточки объявления.
+  // Клонируем имеющийся шаблон, выбираем в новом шаблоне элементы, добавляем информацию.
+  function fillCardWithInformation(cardInfo) {
+    var cardTemplate = document.querySelector('#card').content.querySelector('.map__card');
+    var newCard = cardTemplate.cloneNode(true);
+
+    // Сделаем массив с шаблонами для конкатенации строк.
```

Comment:
```suggestion
    // Сделаем объект с шаблонами для конкатенации строк.
```
массив, это всё, что с квадратными скобками, как скобки фигурные - это объект